pkg/ruler/manager.go

```diff
@@ -98,11 +98,7 @@ func NewDefaultMultiTenantManager(cfg Config, managerFactory ManagerFactory, reg
 }
 
 func (r *DefaultMultiTenantManager) SyncRuleGroups(ctx context.Context, ruleGroups map[string]rulespb.RuleGroupList) {
-	// A lock is taken to ensure if this function is called concurrently, then each call
-	// returns after the call map files and check for updates
-	r.userManagerMtx.Lock()
-	defer r.userManagerMtx.Unlock()
-
+	// This function is never called concurrently for now, so it doesn't need a lock
 	if !r.cfg.TenantFederation.Enabled {
 		RemoveFederatedRuleGroups(ruleGroups)
 	}
```

Comment:
Below there's a `for` loop to "Check for deleted users and remove them" and then we update `r. managersTotal`. We access `r.userManagers` there, so we need the lock.